test/functions/resolution/instantiateMax/instMaxOKifNonrecursive.chpl

```diff
@@ -0,0 +1,12 @@
+proc foo(x) {
+  writeln("In foo with type ", x.type:string);
+}
+
+record R {
+  param i: int;
+}
+
+for param i in 1..257 {
+  var myR = new R(i);
+  foo(myR);
+}
```

Comment:
I suggest making a few more shots over 256, just in case. How about, arbitrarily, 280?

If we accidentally do not have a test for hitting the limit due to actual recursion, please add one.

Reply:
Why do you think 280 is any better of a test than 257 given that the limit is currently 256?

> If we accidentally do not have a test for hitting the limit due to actual recursion, please add one.

We do have such tests which are how I verified that the feature continued working as before.

Reply:
I would like to give it at least a couple of extra steps beyond the limit. Because one could imagine a compiler inlining the first couple of recursive calls without incrementing the counter. Surely not our case, though.

Reply:
I'm not sure what you're referring to:  there's no recursion in this test.

Reply:
I ended up adding a .compopts file that, on one run, set `--instantiate-max` to 230.  This has the benefit of guarding against the test becoming obsolete if the default instantiate-max is raised in the future while also giving you the (IMO, superstitious) delta between instantiate max and how many instantiations the test creates that you were hoping for.